tests/test_topline_historical_backfill.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+
+from click.testing import CliRunner
+
+import pytest
+from mozetl.topline import historical_backfill as backfill
+from pyspark.sql import SparkSession
+
+
+@pytest.fixture(scope="session")
+def spark(request):
+    spark = (SparkSession
+             .builder
+             .appName("test_topline_historical_backfill")
+             .getOrCreate())
+
+    # teardown
+    request.addfinalizer(lambda: spark.stop())
+
+    return spark
+
+
+default_sample = {
+    "geo": "US",
+    "channel": "nightly",
+    "os": "Windows",
+    "date": "2016-01-01",
+    "actives": "1",
+    "hours": "1",
+    "inactives": "1",
+    "new_records": "1",
+    "five_of_seven": "1",
+    "total_records": "1",
+    "crashes": "1",
+    "default": "1",
+    "google": "1",
+    "bing": "1",
+    "yahoo": "1",
+    "other": "1"
+}
+
+
+def generate_samples(snippets, base_sample):
```

Comment:
Done.

This is a slightly modified version of the same function I used in the churn job, so it probably won't be the last time you see it. I think I might end up implementing tests in the same style for `topline_dashboard` and `churn`, so this would be a nice place to start refactoring after everything is in a working state.